### PR TITLE
EVG-12726 route to configure a patch

### PR DESCRIPTION
**Pull request by john-m-liu**
- create a sort-of copy of the MustHaveUser function that is currently in rest/route so that the graphql package can use it without depending on route. Both are short wrappers around gimlet.GetUser so I don't see it as code duplication
- remove a redundant display task struct in the model package and instead replace it with the one in patch
- add a route that configures a patch, which is a wrapper around the existing function called by the UI

### CODE REVIEW

File: rest/route/patch.go [hunk 1]
```diff
@@ -484,3 +487,124 @@ func (p *mergePatchHandler) Run(ctx context.Context) gimlet.Responder {
...
+		if err != nil {
+			return gimlet.MakeJSONErrorResponder(errors.Wrap(err, "unable to find project from patch"))
+		}
+	} else {
+		project, err = dbModel.FindProjectFromVersionID(dbVersion.Id)
+		if err != nil {
+			return gimlet.MakeJSONErrorResponder(errors.Wrap(err, "unable to find project from version"))
+		}
+	}
+	variantTasks := graphql.PatchVariantsTasksRequest{}
```
Comment (ablack12) [inline rest/route/patch.go:575]: Is it weird to be using graphql to be doing completely backend things? Are we worried that having this package depend on graphql could lead to a strange dependency in the future? I think maybe PatchVariantsTasksRequest doesn't belong in graphql, and graphql should import it from a model package or something.

File: rest/route/patch.go [hunk 2]
```diff
@@ -484,3 +487,124 @@ func (p *mergePatchHandler) Run(ctx context.Context) gimlet.Responder {
...
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "unable to get config"))
+	}
+	token, err := settings.GetGithubOauthToken()
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "unable to get token"))
+	}
+	dbVersion, _ := p.sc.FindVersionById(p.patchId)
+	var project *dbModel.Project
+	grip.Warning(p.patch)
```
Comment (ablack12) [inline rest/route/patch.go:563]: debugging logic?

File: rest/route/patch.go [hunk 3]
```diff
@@ -484,3 +487,124 @@ func (p *mergePatchHandler) Run(ctx context.Context) gimlet.Responder {
...
+			return gimlet.MakeJSONErrorResponder(errors.Wrap(err, "unable to find project from patch"))
+		}
+	} else {
+		project, err = dbModel.FindProjectFromVersionID(dbVersion.Id)
+		if err != nil {
+			return gimlet.MakeJSONErrorResponder(errors.Wrap(err, "unable to find project from version"))
+		}
+	}
+	variantTasks := graphql.PatchVariantsTasksRequest{}
+	for i, v := range p.variantTasks.Variants {
```
Comment (ablack12) [inline rest/route/patch.go:576]: This section all reads more awkwardly than necessary I think. Can `patch.VariantTasks{Variant: v.Id}` first be like it's own `newVariant` variable or something, and then you can add to newVariant.Tasks and newVariant.DisplayTasks before appending to `variantTasks.VariantsTasks`?

File: rest/route/patch.go [hunk 4]
```diff
@@ -484,3 +487,124 @@ func (p *mergePatchHandler) Run(ctx context.Context) gimlet.Responder {
...
+	dbVersion, err = p.sc.FindVersionById(versionId)
+	if err != nil {
+		return gimlet.MakeJSONErrorResponder(errors.Wrap(err, "unable to find patch version"))
+	}
+	if dbVersion == nil {
+		return gimlet.NewJSONErrorResponse("no patch found")
+	}
+	restVersion := model.APIVersion{}
+	err = restVersion.BuildFromService(dbVersion)
+	if err != nil {
```
Comment (ablack12) [inline rest/route/patch.go:606]: can you combine with 605?

**Review by ablack12 — CHANGES_REQUESTED** [review]

File: model/project.go [hunk 5]
```diff
@@ -1495,6 +1490,29 @@ func (p *Project) CommandsRunOnBV(cmds []PluginCommandConf, cmd, bv string) []Pl
 	return matchingCmds
 }
 
+func (p *Project) GetVariant(variant string) *BuildVariant {
```
[the remark this answers is not in the document]
Comment (john-m-liu) [reply model/project.go]: doh, I swear I looked beforehand 🐼

File: rest/route/patch.go (hunk 2 above)
Comment (john-m-liu) [reply rest/route/patch.go]: oops removed

File: rest/route/patch.go (hunk 1 above)
Comment (john-m-liu) [reply rest/route/patch.go:565]: a bit, but I don't think it's too awkward. Right now it's logic that was previously in the service package, but is now shared so it's not any worse than what was before. Arguably this specific function should be in model, but for the most part I think it's fine

File: rest/route/patch.go (hunk 3 above)
Comment (john-m-liu) [reply rest/route/patch.go]: yep, done

File: rest/route/patch.go (hunk 4 above)
Comment (john-m-liu) [reply rest/route/patch.go]: done

**Review by ablack12 — APPROVED** [review]

**Pull request merged**